amplify/function/utils/food-groups.ts

```diff
@@ -0,0 +1,125 @@
+}
+
+/**
+ *  Return the start and end calgary time time for a specific group
+ */
+export function getTimeFromGroupNumber(
+  groupNumber: number,
+  groups: number | undefined | null,
+  startTime: string,
+  endTime: string,
+): string {
+  if (groups === undefined || groups == null) groups = 1;
+
+  const start = getCalgaryTime(startTime);
+  const end = getCalgaryTime(endTime);
+
+  //total time between start and end
+  const totalDuration = Math.abs(end.getTime() - start.getTime());
+  const groupDuration = totalDuration / groups;
+
+  const groupStartTime = getCalgaryTime(
+    start.getTime() + groupDuration * groupNumber,
+  );
+  const groupEndTime = getCalgaryTime(
+    start.getTime() + groupDuration * (groupNumber + 1),
+  );
+
+  const options: Intl.DateTimeFormatOptions = {
+    year: "numeric",
+    month: "long",
+    day: "2-digit",
+    hour: "numeric",
+    minute: "numeric",
+    timeZone: "America/Edmonton", // Make sure this is here, we want calgary time
+    hour12: true,
+  };
+
+  // Format the date with the specified options
+  return (
+    groupStartTime.toLocaleString("en-US", options).replace(/,/g, "") +
+    " to " +
+    groupEndTime.toLocaleString("en-US", options).replace(/,/g, "")
+  );
+}
+
+/**
+ * Get the group number (the order which groups can get food) for a user from time
+ * Useful for times when we need to get the current expected group and time slot from the local calgary time
+ */
+export function getGroupNumberFromTime(
+  target: Date,
+  groups: number | undefined | null,
+  startTime: string,
+  endTime: string,
+): number {
+  // Ensure groups is a number and default to 1 if not
+  if (groups === undefined || groups == null) groups = 1;
+  const start = getCalgaryTime(startTime);
+  const end = getCalgaryTime(endTime);
+
+  const totalDuration = Math.abs(end.getTime() - start.getTime());
+  const groupDuration = totalDuration / groups;
+
+  // If the time is before the start time or after the end time, return an invalid group
+  if (target < start || target > end) {
+    return -1;
+  }
+
+  // Calculate the group number for times within the range
+  const timeFromStart = Math.abs(target.getTime() - start.getTime());
+  const groupNumber = Math.floor(timeFromStart / groupDuration);
+
+  return groupNumber;
+}
+
+/**
+ * gets the group number (the order which groups can get food) for a user
+ */
+export function getGroupNumber(
+  userID: string,
+  eventID: string,
+  groups: number | undefined | null,
+) {
+  if (groups === undefined || groups == null) groups = 1;
+
+  return Math.abs((uuidToInteger(userID) + uuidToInteger(eventID)) % groups);
+}
+
+/**
+ *   get the time which a user can eat
+ */
+export function getUserTimeSlot(
+  userID: string,
+  eventID: string,
+  groups: number | undefined | null,
+  startTime: string,
+  endTime: string,
+) {
+  if (groups === undefined || groups == null) groups = 1;
+
+  const userGroupNumber = getGroupNumber(userID, eventID, groups);
```

Comment:
This should be returning the user's Group ID or something, what it's returning right now doesn't seem to help with splitting users by group. At least to my understanding

Reply:
Food Groups are a different thing completely, this is to get the group which will get food